android/app/src/main/kotlin/coop/polypoly/polypod/OnboardingActivity.kt

```diff
@@ -33,14 +45,51 @@ class OnboardingActivity : AppCompatActivity() {
                 R.id.headline_sub to R.string.onboarding_slide3_sub_headline,
                 R.id.body_text to R.string.onboarding_slide3_body_text,
             ),
+            mapOf(
+                R.id.headline_main to R.string.onboarding_slide4_headline,
+                R.id.headline_sub to R.string.onboarding_slide4_sub_headline,
+                R.id.body_text to R.string.onboarding_slide4_body_text,
+            ),
         )
 
+        if (!Preferences.isFirstRun(baseContext)) {
+            strings = listOf(strings[2])
+        }
+
         carousel.pageCount = strings.size
-        carousel.setViewListener { position ->
+        carousel.setViewListener { requestedPosition ->
+            var position = requestedPosition
+            if (position == 2 && shouldSkipBiometricsPrompt()) {
+                position ++
+                if (position > carousel.pageCount) {
+                    close()
+                }
+                carousel.setCurrentItem(position)
+            }
+
             val slide = layoutInflater.inflate(R.layout.onboarding_slide, null)
             strings[position].forEach { (viewId, stringId) ->
                 slide.findViewById<TextView>(viewId).text = getString(stringId)
             }
+            if ((position == strings.size - 2) || (strings.size == 1)) {
```

Comment:
Here you could probably just retrieve the slide at the respective position and check if it's the one you want.

Reply:
Done